pyaerocom/ungriddeddata_structured.py

```diff
@@ -639,21 +675,22 @@ def append_station_data(
                 if var not in self.var_idx:
                     self.var_idx[var] = len(self.var_idx)
                 var_idx = self.var_idx[var]
-                self.metadata[meta_idx]["var_info"] = {}
-                self.metadata[meta_idx]["var_info"][var] = {}
 
-                self.metadata[meta_idx]["var_info"][var].update(station_data["var_info"][var])
+                vi[var].update(station_data["var_info"][var])
                 for x in ("longitude", "latitude", "altitude"):
-                    if x not in self.metadata[meta_idx]["var_info"][var]:
-                        self.metadata[meta_idx]["var_info"][var][x] = station_data[x]
+                    if x not in vi[var]:
+                        vi[x] = station_data[x]
 
-                uds = UngriddedDataStructured(num_points=len(values))
+                if uds is None:
+                    uds = UngriddedDataStructured(num_points=len(values))
                 v_data = uds._dra._array  # access to raw numpy-array
                 v_data["meta_id"][:] = meta_idx
                 v_data["var_id"][:] = var_idx
                 v_data["data"] = values
                 v_data["start_time"] = times
                 # v_data["end_time"] not used
+                if altitude is not None:
+                    v_data["dataaltitude"] = altitude
                 # v_data["dataaltitude"] not used
```

Comment:
@lewisblake looks like you started using `v_data["dataaltitude"]` for vertical profiles, so the comment should be removed or modified to `v_data["dataaltitude"] used for vertical profiles`